breacharbiter.go

```diff
@@ -827,73 +882,100 @@ type retributionInfo struct {
 	doneChan chan struct{}
 }
 
+// newRetributionInfo constructs a retributionInfo containing all the
+// information required by the breach arbiter to recover funds from breached
+// channels.  The information is primarily populated using the BreachRetribution
+// delivered by the wallet when it detects a channel breach.
+func newRetributionInfo(chanPoint *wire.OutPoint,
+	breachInfo *lnwallet.BreachRetribution,
+	chanInfo *channeldb.ChannelSnapshot) *retributionInfo {
+
+	// First, record the breach information and witness type for the local
+	// channel point.  This will allow us to completely generate a valid
+	// witness in the event of failures, as it will be persisted in the
+	// retribution store.  Here we use CommitmentNoDelay since
+	selfOutput := newBreachedOutput(
+		&breachInfo.LocalOutpoint,
+		lnwallet.CommitmentNoDelay,
+		&breachInfo.LocalOutputSignDesc,
+	)
+
+	// Second, record the same information and witness type regarding the
+	// remote outpoint, which belongs to the party who tried to steal our
+	// money! Here we set witnessType of the breachedOutput to
+	// CommitmentRevoke, since we will be using a revoke key, and
+	// withdrawing the funds from the commitment transaction immediately.
+	revokedOutput := newBreachedOutput(
+		&breachInfo.RemoteOutpoint,
+		lnwallet.CommitmentRevoke,
+		&breachInfo.RemoteOutputSignDesc,
+	)
+
+	// Determine the number of second layer HTLCs we will attempt to sweep.
+	nHtlcs := len(breachInfo.HtlcRetributions)
+
+	// Lastly, for each of the breached HTLC outputs, assemble the
+	// information we will persist to disk, such that we will be able to
+	// deterministically generate a valid witness for each output. This will
+	// allow the breach arbiter to recover from failures, in the event that
+	// it must sign and broadcast the justice transaction.
+	var htlcOutputs = make([]*breachedOutput, nHtlcs)
```

Comment:
Typically I use `var` to alert the reader to the fact that the variable is mutable, which helps to logically categorize them (especially in debugging+review) from static references, e.g. `:=`. Most often, the scope in which a variable is modified is limited to the following stanza, which permits the reader to assume that following stanza modifies or constructs the mutable variable being introduced. Even if the variable is modified much later in the function, the reader is at least alerted to which variables are intended to be altered dynamically. Counterintuitively, this in fact gives greater confidence as to which variables can be assumed to be static, which reduces working complexity and cognitive load on behalf of the reader. Otherwise, one more or less has to assume that any variable assigned with `:=` could be mutable. There are certainly exceptions, but in general I've found it to be a very helpful form of self-documentation. Happy to change if we prefer the brevity of define+assign.